### PR TITLE
Reference smoothr

### PR DESCRIPTION
**Pull request by Lvulis**
Per #830, I've added some references to `smoothr` in Chapters 5 and 6 and updated the figure of the US states. Also fixed sentences in the introduction of Ch5.

### CODE REVIEW

File: 05-geometry-operations.Rmd [hunk 1]
```diff
@@ -101,14 +101,30 @@ us_states_simp2 = rmapshaper::ms_simplify(us_states2163, keep = 0.01,
                                           keep_shapes = TRUE)
 ```
 
-Finally, the visual comparison of the original dataset and the two simplified versions shows differences between the Douglas-Peucker (`st_simplify`) and Visvalingam (`ms_simplify`) algorithm outputs (Figure \@ref(fig:us-simp)):
 
-```{r us-simp, echo=FALSE, fig.cap="Polygon simplification in action, comparing the original geometry of the contiguous United States with simplified versions, generated with functions from sf (center) and rmapshaper (right) packages.", warning=FALSE, fig.asp=0.3, fig.scap="Polygon simplification in action."}
+An alternative to simplification is smoothing the boundaries of polygon and linestring geometries, which is implemented in the **smoothr** package. 
+Smoothing interpolates the edges of geometries and does not necessarily lead to fewer vertices, but can beespecially useful when working with geometries that arise from spatially vectorizing a raster (a topic covered in Chapter \@ref(raster-vector).
```
Comment (Nowosad) [inline 05-geometry-operations.Rmd:106]: typo "beespecially"

File: 05-geometry-operations.Rmd [hunk 2]
```diff
@@ -27,7 +27,7 @@ Section \@ref(geo-ras) covers geometric transformations on raster objects.
 This involves changing the size and number of the underlying pixels, and assigning them new values.
 It teaches how to change the resolution (also called raster aggregation and disaggregation), the extent and the origin of a raster.
 These operations are especially useful if one would like to align raster datasets from diverse sources.
-Aligned raster objects share a one-to-one correspondence between pixels, allowing them to be processed using map algebra operations, described in Section \@ref(map-algebra). The final Section \@ref(raster-vector) connects vector and raster objects. 
+Aligned raster objects share a one-to-one correspondence between pixels, allowing them to be processed using map algebra operations, described in Section \@ref(map-algebra). The interaction between raster and vector objects is covered in Chapter \@ref(raster-vector).
```
Comment (Nowosad) [inline 05-geometry-operations.Rmd:30]: Please move this sentence to a new line

File: 05-geometry-operations.Rmd [hunk 3]
```diff
@@ -101,14 +101,30 @@ us_states_simp2 = rmapshaper::ms_simplify(us_states2163, keep = 0.01,
...
+**smoothr** implements three techniques for smoothing: a Gaussian kernel regression, Chaikin's corner cutting algorithm, and spline interpolation, which are all described in the package vignette and [website](https://strimas.com/smoothr/). 
+Note that similar to `st_simplify()`, the smoothing algorithms don't preserve 'topology'.
+
+The workhorse function of **smoothr** is `smooth()`, where the `method` argument specifies what smoothing technique to use.
+Below is an example of using Gaussian kernel regression to smooth the borders of US states by using `method=ksmooth`.
+The `smoothness` argument controls the bandwidth of the Gaussian that is used to smooth the geometry and has a default value of 1.
+
+```{r 05-geometry-operations-6b, warning=FALSE}
+us_states_simp3 = smoothr::smooth(us_states2163, method = 'ksmooth', smoothness = 6)
+
```
Comment (Nowosad) [inline 05-geometry-operations.Rmd:116]: remove empty line

File: 05-geometry-operations.Rmd [hunk 4]
```diff
@@ -101,14 +101,30 @@ us_states_simp2 = rmapshaper::ms_simplify(us_states2163, keep = 0.01,
...
+```
+
+Finally, the visual comparison of the original dataset, the two simplified versions, and the smoothed version shows differences between the Douglas-Peucker (`st_simplify`) and Visvalingam (`ms_simplify`) algorithm outputs (Figure \@ref(fig:us-simp)):
```
Comment (Nowosad) [inline 05-geometry-operations.Rmd:119]: Please try to improve this sentence or split it into two sentences
.

File: 06-raster-vector.Rmd [hunk 5]
```diff
@@ -479,6 +479,10 @@ grain_poly = as.polygons(grain) |>
   st_as_sf()
 ```
 
+The aggregated polygons of the `grain` dataset have rectilinear boundaries which arise from being defined by the boundaries of pixels.
```
Comment (Nowosad) [inline 06-raster-vector.Rmd:482]: "the boundaries of pixels" -> "the shapes of pixels"?

File: 06-raster-vector.Rmd [hunk 6]
```diff
@@ -479,6 +479,10 @@ grain_poly = as.polygons(grain) |>
...
+The **smoothr** package described in Chapter \@ref(geometry-operations) can be used to smooth the edges of the polygons.
+The smoothed polygons will not have the same exact spatial footprint as the original pixels, so caution should be used for using the smoothed polygons for further analysis.
```
Comment (Nowosad) [inline 06-raster-vector.Rmd:484]: What does the "spatial footprint" mean?

File: 05-geometry-operations.Rmd (hunk 2 above)
Comment (Lvulis) [reply 05-geometry-operations.Rmd:30]: Fixed

File: 05-geometry-operations.Rmd (hunk 1 above)
Comment (Lvulis) [reply 05-geometry-operations.Rmd:106]: Fixed

File: 06-raster-vector.Rmd (hunk 6 above)
Comment (Lvulis) [reply 06-raster-vector.Rmd]: Changed this sentence to be something more like "coverage" and made it more explicit, maybe this is clearer?

**Review by Lvulis — COMMENTED** [review]
See updated commit with revised text.

**Comment by Nowosad** [comment]
Great! Thank you a lot

**Pull request merged**